src/univariate/affine.jl

```diff
@@ -0,0 +1,182 @@
+@deprecate LocationScale(args...; kwargs...) affine(args...)
+
+"""
+    affine(shift, scale, base_dist)
+
+Return a shifted and scaled (affinely transformed) version of `base_dist`.
+
+If ``Z`` is a random variable with distribution `base_dist`, then `affine` returns the 
+distribution of the random variable
+```math
+X = shift + scale * Z
+```
+
+If `base_dist` is a discrete distribution, the probability mass function of the transformed 
+distribution is given by
+```math
+P(X = x) = P\\left(Z = \\frac{x-shift}{scale} \\right).
+```
+If `base_dist` is a continuous distribution, the probability density function of
+the transformed distribution is given by
+```math
+f(x) = \\frac{1}{scale} base_dist \\! \\left( \\frac{x-shift}{scale} \\right).
+```
+
+If `base_dist` falls under a location-scale family, `affine` will attempt to return a 
+transformed distribution of type `base_dist` if an `affine` method has been implemented. 
+Otherwise, it will fall back on the `Affine` wrapper type.
+
+```julia
+affine(shift, scale, base_dist)  # location-scale transformed distribution
+params(d)            # Get the parameters, i.e. (shift, scale, base_dist)
+location(d)          # Get the location parameter
+scale(d)             # Get the scale parameter
+```
+
+External links
+[Location-Scale family on Wikipedia](https://en.wikipedia.org/wiki/Location%E2%80%93scale_family)
+"""
+struct Affine{T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}} <: UnivariateDistribution{S}
+    shift::T
+    scale::T
+    base_dist::D
+    function Affine{T,S,D}(shift::T, scale::T, base_dist::D; check_args=true) where {
+        T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}
+    }
+        !check_args || @check_args(Affine, !iszero(scale))
+        return new{T,S,D}(shift, scale, base_dist)
+    end
+end
+
+function affine(shift::Real, scale::Real, base_dist::UnivariateDistribution)
+    T = promote_type(eltype(base_dist), typeof(shift), typeof(scale))
+    D = typeof(base_dist)
+    S = value_support(D)
+    return Affine{T,S,D}(T(shift), T(scale), base_dist)
+end
+
+# Composing affine transformations
+function affine(shift::Real, scale::Real, d::Affine)
+    return affine(shift + scale * d.shift, scale * d.scale, d.base_dist)
+end
+
+reflected(d::Affine) = d.scale < 0
+
+# Aliases
+const ContinuousAffine{T<:Real,D<:ContinuousUnivariateDistribution} = Affine{T,Continuous,D}
+const DiscreteAffine{T<:Real,D<:DiscreteUnivariateDistribution} = Affine{T,Discrete,D}
+
+Base.eltype(::Type{<:Affine{T}}) where {T} = T
```

Comment:
This is wrong (see the `eltype` PR for a discussion).

Reply:
@devmotion Hmm... would `Base.eltype(::Type{<:AffineDistribution{T, S, D}}) where {T, S, D} = promote(T, eltype(D))` be what we'd want?

Reply:
Only for discrete distributions and only when `eltype(D)` *has* to be promoted - e.g. if it is `Int` but the location or scale are of type `Rational` or `Float64`. In general, the `eltype` should not depend on the type of the parameters.

Reply:
@devmotion won't the return values for `rand(Affine)` always be `promote(T, eltype(D))`? since we generate them as `shift + scale * rand(D)`, which will automatically promote

Reply:
Currently, this is the case for `LocationScale` but this is not correct - the output of `rand` should not depend on the parameters in general. There's much more explanations in the PR.